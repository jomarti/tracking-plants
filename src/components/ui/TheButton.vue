<template>
  <button :class="buttonClasses">
    <slot />
  </button>
</template>

<script>
export default {
  name: 'TheButton',
  props: {
    theme: {
      type: String,
      default: 'primary',
      validate: value => ['primary'].indexOf(value) !== -1
    },
    size: {
      type: String,
      default: 'medium',
      validate: value => ['small', 'medium', 'large'].indexOf(value) !== -1
    }
  },
  computed: {
    buttonClasses() {
      const commonClasses =
        'border rounded-full py-2 px-6 font-bold focus:outline-none';
      const primaryClasses = `${commonClasses} bg-green-600 text-white`;

      const mediumClasses = 'w-3/5';
      const largeClasses = 'w-full';

      return {
        [primaryClasses]: this.theme === 'primary',
        [mediumClasses]: this.size === 'medium',
        [largeClasses]: this.size === 'large'
      };
    }
  }
};
</script>
