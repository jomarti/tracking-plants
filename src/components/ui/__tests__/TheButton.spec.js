import { shallowMount } from '@vue/test-utils';
import TheButton from '../TheButton';

describe('Global Components -> ui -> TheButton', () => {
  const build = propsData => {
    const options = propsData ? { propsData } : null;
    const wrapper = shallowMount(TheButton, options);
    return {
      wrapper
    };
  };
  it('should match snapshot', () => {
    const { wrapper } = build();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should add the primary theme classes when theme prop is primary', () => {
    const { wrapper } = build({ theme: 'primary' });

    const button = wrapper.find('button');
    const buttonClasses = button.classes();
    expect(buttonClasses).toContain('bg-green-600');
    expect(buttonClasses).toContain('text-white');
  });

  it('should add the small size classes when size prop is small', () => {
    const { wrapper } = build({ size: 'small' });

    const button = wrapper.find('button');
    const buttonClasses = button.classes();
    expect(buttonClasses).not.toContain('w-3/5');
    expect(buttonClasses).not.toContain('w-full');
  });

  it('should add the medium size classes when size prop is medium', () => {
    const { wrapper } = build({ size: 'medium' });

    const button = wrapper.find('button');
    const buttonClasses = button.classes();
    expect(buttonClasses).toContain('w-3/5');
  });

  it('should add the large size classes when size prop is large', () => {
    const { wrapper } = build({ size: 'large' });

    const button = wrapper.find('button');
    const buttonClasses = button.classes();
    expect(buttonClasses).toContain('w-full');
  });
});
