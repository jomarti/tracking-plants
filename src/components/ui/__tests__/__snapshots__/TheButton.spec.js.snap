// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Components -> ui -> TheButton should match snapshot 1`] = `
<button
  class="border rounded-full py-2 px-6 font-bold focus:outline-none bg-green-600 text-white w-3/5"
/>
`;
